django/core/management/commands/shell.py

```diff
@@ -43,22 +46,35 @@ def add_arguments(self, parser):
                 "exit."
             ),
         )
+        parser.add_argument(
+            "-d",
+            "--display",
+            action="store_true",
+            default=False,
+            dest="display",
+            help=("Display loaded models messages"),
+        )
 
     def ipython(self, options):
         from IPython import start_ipython
 
-        start_ipython(argv=[])
+        try:
+            imported_objects = self.get_namespace(options, self.style)
+            start_ipython(argv=[], user_ns=imported_objects)
+        except ImportError:
+            traceback.format_exc()
```

Comment:
Using `try / except` without a `raise` will cause this Python process to exit with return code zero, meaning “success”, even when it should crash an error.

Also, isn’t using `traceback.format_exc()` basically the same as what Python does when exiting with an exception? If so, we can simplify to just allow the crash:

Isn’t `traceback.format_exc()` the same 
```suggestion
        imported_objects = self.get_namespace(options, self.style)
        start_ipython(argv=[], user_ns=imported_objects)
```